### ISSUE TITLE
Make 'mapping' variables 'const'.

### ISSUE THREAD

**Issue by bangerth** [dealii/dealii#17260]
Mapping variables should always be `const` because they provide information to `FEValues` and related classes, but do not actually change state. So if we see
```
  MappingQ<dim> mapping;
```
somewhere in a class declaration, we might as well make that
```
  const MappingQ<dim> mapping;
```

Here is a list of places where one could make this kind of change:
```
examples/step-11/step-11.cc:87:    MappingQ<dim>      mapping;
examples/step-16/step-16.cc:376:    MappingQ1<dim> mapping;
examples/step-16/step-16.cc:428:    MappingQ1<dim>     mapping;
examples/step-34/step-34.cc:236:    MappingQ<dim - 1, dim>      mapping;
examples/step-37/step-37.cc:691:    MappingQ1<dim> mapping;
examples/step-38/step-38.cc:117:    MappingQ<dim, spacedim>      mapping;
examples/step-44/step-44.cc:3222:    MappingQEulerian<dim> q_mapping(degree, dof_handler, soln);
examples/step-45/step-45.cc:106:    MappingQ<dim> mapping;
examples/step-47/step-47.cc:164:    MappingQ<dim> mapping;
examples/step-48/step-48.cc:312:    MappingQ1<dim> mapping;
examples/step-59/step-59.cc:929:    MappingQ1<dim> mapping;
examples/step-64/step-64.cc:276:    MappingQ<dim> mapping(fe_degree);
examples/step-65/step-65.cc:613:      MappingQ<dim> mapping(fe.degree + 1);
examples/step-67/step-67.cc:1776:    MappingQ<dim>       mapping;
examples/step-68/step-68.cc:234:    MappingQ1<dim>                             mapping;
examples/step-6/step-6.cc:477:    MappingQ<dim> mapping(3);
examples/step-76/step-76.cc:1206:    MappingQ<dim>       mapping;
examples/step-79/step-79.cc:755:    MappingQ<dim>         mapping(1);
examples/step-79/step-79.cc:1303:    MappingQ<dim>         mapping(1);
examples/step-79/step-79.cc:1629:      MappingQ<dim>         mapping(1);
examples/step-87/step-87.cc:164:    MappingQ1<dim>     mapping;
examples/step-87/step-87.cc:302:    MappingQ1<dim>                mapping;
examples/step-87/step-87.cc:469:    MappingQ1<dim>                mapping;
examples/step-87/step-87.cc:739:    MappingQ1<dim>      mapping_background;
examples/step-87/step-87.cc:761:    MappingQ<dim - 1, dim>      mapping_immersed_base(3);
```

**Comment by masterleinad** [comment]
Note that making a member variable `const` implicitly deletes the copy assignment operator.

**Comment by bangerth** [comment]
True, but I think that for the main classes of tutorials, that's ok. I've already made the finite element objects `const` in #16968.

**Comment by gassmoeller** [comment]
This was addressed in #17392
